src/lib/CMakeLists.txt

```diff
@@ -244,6 +244,18 @@ set_target_properties(librnp
 
 if (BUILD_SHARED_LIBS)
   add_library(librnp-static STATIC $<TARGET_OBJECTS:librnp-obj> $<TARGET_OBJECTS:rnp-common>)
+  # Limit symbols export only to rnp_* functions.
+  if (APPLE)
+    # Check for Apple OSs and use -export_symbols_list
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols" "_rnp_*\n")
+    target_link_options(librnp PRIVATE -Wl,-exported_symbols_list "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols")
+    set_target_properties(librnp PROPERTIES LINK_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols")
+  elseif(NOT WIN32)
+    # Use --version-script otherwise
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/librnp.vsc" "{\nglobal: rnp_*;\nlocal: *;\n};\n")
```

Comment:
Is there a reason to write this file via cmake `file(WRITE[…])` instead of just storing `librnp.vsc` directly in the source?

Tracking `librnp.vsc` in the source explicitly would make it easier to add things like version tags to new symbols going forward.

Reply:
The only reason is that librnp.vsc/librnp.symbols are almost one-liners, so it looked simpler to write it out from CMake instead of adding the file. However, probably it would be more clear to have those files in the source tree, agree.